wayland-protocols/mir-shell-unstable-v1.xml

```diff
@@ -0,0 +1,452 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<protocol name="mir_shell_unstable_v1">
+  <copyright>
+    Copyright © 2023 Canonical Limited
+
+    Permission to use, copy, modify, distribute, and sell this
+    software and its documentation for any purpose is hereby granted
+    without fee, provided that the above copyright notice appear in
+    all copies and that both that copyright notice and this permission
+    notice appear in supporting documentation, and that the name of
+    the copyright holders not be used in advertising or publicity
+    pertaining to distribution of the software without specific,
+    written prior permission.  The copyright holders make no
+    representations about the suitability of this software for any
+    purpose.  It is provided "as is" without express or implied
+    warranty.
+
+    THE COPYRIGHT HOLDERS DISCLAIM ALL WARRANTIES WITH REGARD TO THIS
+    SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
+    FITNESS, IN NO EVENT SHALL THE COPYRIGHT HOLDERS BE LIABLE FOR ANY
+    SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+    AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+    ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+    THIS SOFTWARE.
+  </copyright>
+
+  <interface name="mir_shell_v1" version="1">
+    <description summary="create surface archetypes that are recognised by the desktop">
+      Clients can use this interface to assign an archetype to wl_surfaces.
+    </description>
+
+    <request name="get_regular_surface">
+      <description summary="assign the mir_regular_surface_v1 archetype">
+        Create a mir_regular_surface_v1 archetype for the given surface. This assigns
+        the archetype of regular_surface to the wl_surface, or raises a protocol error
+        if another archetype is already assigned and the transition is disallowed.
```

Comment:
How does a client know what transitions are allowed?

Reply:
I need to document a lot more of this section of "Mir and Unity: Surfaces, input, and displays" somewhere. I put off doing that because I felt it more urgent to get this discussion going.

Reply:
Ah, good. I wasn't sure if the plan was to dump a bunch of that in here.